### ISSUE TITLE
Raw identifiers act as a comment in the `extendr-wrappers.R` file

### ISSUE THREAD

**Issue by bicarlsen** [extendr/extendr#528]
If a [raw identifier](https://doc.rust-lang.org/rust-by-example/compatibility/raw_identifiers.html) is used as a function argument, the `#` is copied in the `extendr-wrappers.R` file, acting as a comment.
e.g.
```rust
#[extendr]
fn my_fcn(r#type: i32) -> i32 {
  ...
}

extendr_module! {
  mod my_lib;
  my_fcn;
}
```
results in the output line `my_fcn <- function(r#type) .Call(wrap__my_fcn, r#type)` in the `extendr-wrappers.R` file.

**Comment by Ilia-Kosenkov** [comment]
Hi! Thanks for reporting this. Yep, I can totally see why it happens like this. We will fix it soon-ish.

**Comment by CGMossa** [comment]
I'm sure @Ilia-Kosenkov will help you out in a minute. Note if you want to chat with us or ask informally about things, some of us hang out on the project Discord sever.

**Comment by Ilia-Kosenkov** [comment]
Oh, cool. I already researched the issue and prototype some fixes, but I am more than happy to guide you through the process!
Here is a rough outline of the first steps that you need to do (let's pretend we follow some TDD):

1. Get the extendr repo locally, go to `tests/extendrtests` and ensure you can test this R project
  - By test I mean you can execute `devtools::test()` and all things pass (or do it within Rstudio or any other tool that you use)
  - There is a caveat -- it may fail because it cannot find extendr-api. You need to tweak `Cargo,toml` in `tests/textendrtest/src/rust` so it points to your local `extendr-api` folder (we use relative paths there, you likely need to add/remove a couple of `../` until it matches the path). That way you will be able to test the changes you make to the `extendr-*` projects
2. Once current tests work, add a new rust module in this test project and write there methods that have `r#` raw identifiers. Start with something similar to your example (but give it better test names).
  - Don't forget to reference this new module in `tests/extendrtests/src/rust/src/lib.rs` so you get some bindings generated for it
3. Run `rextendr::document()` over `tests/extendrtest`. It should fail with an error similar to the one above -- the generated bindings will contain invalid R syntax.

**Comment by CGMossa** [comment]
It's worth it for us to get you up and running lol.
You're more than welcome here.

**Comment by Ilia-Kosenkov** [comment]
I agree that if you want to get some deeper understanding of `extendr`, it is well worth of our time to walk you through.

**Comment by bicarlsen** [comment]
@Ilia-Kosenkov after cloning the `master` branch and adjusting the `Cargo.toml` file, `devtools::test()` returns a failing test. I don't think this affects what I'm adding but just wanted to make you aware.
```
> devtools::test()
ℹ Testing extendrtests
✔ | F W S  OK | Context
✔ |        11 | classes                                  
✔ |         4 | graphic-device                           
✔ |         9 | na                                       
✔ |         2 | ndarray                                  
✖ | 1       0 | optional-either                          
─────────────────────────────────────────────────────────
Error (test-optional-either.R:1:1): (code run outside of `test_that()`)
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called ‘patrick’
Backtrace:
 1. base::loadNamespace(x)
      at test-optional-either.R:1:0
 2. base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 3. base (local) withOneRestart(expr, restarts[[1L]])
 4. base (local) doWithOneRestart(return(expr), restart)
─────────────────────────────────────────────────────────
✔ |         2 | test-submodules                          
✔ |        49 | type-conversion [0.2s]                   
✔ |        18 | vector-wrappers                          
✔ |        16 | wrappers                                 

══ Results ══════════════════════════════════════════════
Duration: 0.5 s

[ FAIL 1 | WARN 0 | SKIP 0 | PASS 111 ]
```

**Comment by Ilia-Kosenkov** [comment]
Oh, you need to install that package for tests.

**Comment by CGMossa** [comment]
This means we should add this package to `Suggests`.

**Comment by Ilia-Kosenkov** [comment]
It is there already. https://github.com/extendr/extendr/blob/master/tests/extendrtests/DESCRIPTION#L22

**Comment by CGMossa** [comment]
I swear I checked before messaging.
But according to R Extensions packages for vignettes and tests are in
`Suggests`.
So to dev a package one needs to install `Suggests`..
